catalog/resource_metastore_data_access.go

```diff
@@ -54,93 +49,169 @@ type DataAccessConfiguration struct {
+					})
+				}
+				if err != nil {
+					return err
+				}
+				p.Pack(d)
+				return nil
+			})
 		},
 		Read: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
-			metastoreID, dacID, err := p.Unpack(d)
-			if err != nil {
-				return err
-			}
-			dac, err := NewDataAccessConfigurationsAPI(ctx, c).Get(metastoreID, dacID)
-			if err != nil {
-				return err
-			}
-			metastore, err := NewMetastoresAPI(ctx, c).getMetastore(metastoreID)
+			metastoreId, dacName, err := p.Unpack(d)
 			if err != nil {
 				return err
 			}
-			isDefault := metastore.DefaultDacID == dacID
-			d.Set("is_default", isDefault)
-			return common.StructToData(dac, s, d)
+			var metastore *catalog.MetastoreInfo
+
+			return c.WorkspaceOrAccountRequest(func(acc *databricks.AccountClient) error {
+				var storageCredential *catalog.AccountsStorageCredentialInfo
+				storageCredential, err = acc.StorageCredentials.Get(ctx, catalog.GetAccountStorageCredentialRequest{
+					MetastoreId: metastoreId,
+					Name:        dacName,
+				})
+				if err != nil {
+					return err
+				}
+				m, err := acc.Metastores.GetByMetastoreId(ctx, metastoreId)
+				metastore = m.MetastoreInfo
+				if err != nil {
+					return err
+				}
+				isDefault := metastore.StorageRootCredentialName == dacName
+				d.Set("is_default", isDefault)
+				return common.StructToData(storageCredential, dacSchema, d)
+			}, func(w *databricks.WorkspaceClient) error {
+				var storageCredential *catalog.StorageCredentialInfo
+				storageCredential, err = w.StorageCredentials.GetByName(ctx, dacName)
+				if err != nil {
+					return err
+				}
+				metastore, err = w.Metastores.GetById(ctx, metastoreId)
+				if err != nil {
+					return err
+				}
+				isDefault := metastore.StorageRootCredentialName == dacName
+				d.Set("is_default", isDefault)
+				return common.StructToData(storageCredential, dacSchema, d)
+			})
 		},
 		Delete: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
-			metastoreID, dacID, err := p.Unpack(d)
+			metastoreId, dacName, err := p.Unpack(d)
 			if err != nil {
 				return err
 			}
-			return NewDataAccessConfigurationsAPI(ctx, c).Delete(metastoreID, dacID)
+			return c.WorkspaceOrAccountRequest(func(acc *databricks.AccountClient) error {
+				return acc.StorageCredentials.Delete(ctx, catalog.DeleteAccountStorageCredentialRequest{
+					MetastoreId: metastoreId,
+					Name:        dacName,
+				})
+			}, func(w *databricks.WorkspaceClient) error {
+				return w.StorageCredentials.DeleteByName(ctx, dacName)
+			})
 		},
 	}.ToResource()
 }
+
+// migrate to v1 state, as the id is now changed
+func dacMigrateV0(ctx context.Context,
+	rawState map[string]any,
+	meta any) (map[string]any, error) {
+	newState := map[string]any{}
+	for k, v := range rawState {
+		switch k {
+		case "id":
+			log.Printf("[INFO] Upgrade id")
+			newState[k] = fmt.Sprintf("%v%s%v", rawState["metastore_id"], "|", rawState["name"])
```

Comment:
Related to https://github.com/databricks/terraform-provider-databricks/pull/2528#pullrequestreview-1589264459: do we put metastore_id first in the state or second?